Sources/Amplitude/Amplitude.m

```diff
@@ -284,6 +271,71 @@ - (instancetype)initWithInstanceName:(NSString *)instanceName {
     return self;
 }
 
++ (NSString *)getDataStorageKey:(NSString *)key instanceName:(NSString *)instanceName {
+    NSString *bundleIdentifier = [[NSBundle mainBundle] bundleIdentifier];
+    NSString *dataStorageKey = [NSString stringWithFormat:@"%s_%@_%@_%@", "amplitude", bundleIdentifier, instanceName, key];
+    return dataStorageKey;
+}
+
+- (void)migrateToFileStorage {
+    if ([AMPStorage hasFileStorage:self.instanceName] || [AMPDatabaseHelper hasDatabase:self.instanceName])
+        return;
+
+    int eventCount = [self.dbHelper getEventCount];
+    if (eventCount > 0) {
+        NSMutableArray *events = [self.dbHelper getEvents:-1 limit:-1];
+        self->_eventsBuffer = [events mutableCopy];
+        for(NSMutableDictionary *event in events) {
+            NSError *error = nil;
+            NSData *jsonData = [NSJSONSerialization dataWithJSONObject:[AMPUtils makeJSONSerializable:event] options:0 error:&error];
+            if (error != nil) {
+                AMPLITUDE_ERROR(@"ERROR: could not JSONSerialize event type: %@", error);
+                continue;
+            }
+            NSString *jsonString = [[NSString alloc] initWithData:jsonData encoding:NSUTF8StringEncoding];
+            [AMPStorage storeEvent:jsonString instanceName:self.instanceName];
+        }
+       long long  maxEventId = [[[events lastObject] objectForKey:@"event_id"] longLongValue];
+       [self.dbHelper removeEvents:maxEventId];
+    }
+
+    int identifyCount = [self.dbHelper getIdentifyCount];
+    if (identifyCount > 0) {
+        NSMutableArray *identifys = [self.dbHelper getIdentifys:-1 limit:-1];
+        self->_identifyBuffer = [identifys mutableCopy];
+        for (NSDictionary *event in identifys) {
+            // convert event dictionary to JSON String
+            NSError *error = nil;
+            NSData *jsonData = [NSJSONSerialization dataWithJSONObject:[AMPUtils makeJSONSerializable:event] options:0 error:&error];
+            if (error != nil) {
+                AMPLITUDE_ERROR(@"ERROR: could not JSONSerialize event: %@", error);
+                continue;
+            }
+            NSString *jsonString = [[NSString alloc] initWithData:jsonData encoding:NSUTF8StringEncoding];
+            if ([AMPUtils isEmptyString:jsonString]) {
+                AMPLITUDE_ERROR(@"ERROR: JSONSerializing identify resulted in an NULL string");
+                continue;
+            }
+            [AMPStorage storeIdentify:jsonString instanceName:self.instanceName];
+        }
+        long long maxIdentifyId = [[[identifys lastObject] objectForKey:@"event_id"] longLongValue];
+        [self.dbHelper removeIdentifys:maxIdentifyId];
+    }
+
+    if (eventCount > 0 || identifyCount > 0) {
+        [self uploadEvents];
```

Comment:
Just notice we are calling `uploadEvents` now, would it send the events actually? Or are we writing the same events to the file again?